showy.typ

```diff
@@ -1,197 +1,323 @@
+  let inset = (
+    left: showy-inset(left, inset),
+    right: showy-inset(right, inset)
+  )
+  let (start, end) = (0%, 0%)
+
+  // For relative insets the original width needs to be calculated
+  if type(inset.left) == "ratio" and type(inset.right) == "ratio" {
+    let full = 100% / (1 - float(inset.right) - float(inset.left))
+    start = -inset.left * full
+    end = full + start
+  } else if type(inset.left) == "ratio" {
+    let full = (100% + inset.right) / (1 - float(inset.left))
+    (start, end) = (-inset.left * full, 100% + inset.right)
+  } else if type(inset.right) == "ratio" {
+    let full = (100% + inset.left) / (1 - float(inset.right))
+    (start, end) = (-inset.left, full - inset.left)
+  } else {
+    (start, end) = (-inset.left, 100% + inset.right)
+  }
+
+  line.with(
+    start: (start, 0%),
+    end: (end, 0%)
+  )
+}
+/*
+ * Function: showy-stroke()
+ *
+ * Description: Creates a stroke ot set of strokes
+ * to use as borders.
+ *
+ * Parameters:
+ * + frame: The dictionary with frame settings
+ */
+#let showy-stroke( frame, ..overrides ) = {
+  let (paint, dash, width) = (
+    frame.at("border-color", default: black),
+    frame.at("dash", default: "solid"),
+    frame.at("width", default: 1pt)
+  )
+
+  let strokes = (:)
+  if type(width) != "dictionary" { // Set all borders at once
+    for side in ("top", "bottom", "left", "right") {
+      strokes.insert(side, (paint: paint, dash: dash, thickness: width))
+    }
+  } else { // Set each border individually
+    for pair in width {
+      strokes.insert(
+        pair.first(), // key
+        (paint: paint, dash: dash, thickness: pair.last())
+      )
+    }
+  }
+  for pair in overrides.named() {
+    strokes.insert(
+      pair.first(),
+      (paint: paint, dash: dash, thickness: pair.last())
+    )
+  }
+  return strokes
+}
+
+/*
+ * Function: showybox()
+ *
+ * Description: Creates a showybox
+ *
+ * Parameters:
+ * - frame:
+ *   + upper-color: Color used as background color where the title goes
+ *   + lower-color: Color used as background color where the body goes
+ *   + border-color: Color used for the showybox's border
+ *   + radius: Showybox's radius
+ *   + width: Border width of the showybox
+ *   + dash: Showybox's border style
+ * - title-style:
+ *   + color: Text color
+ *   + weight: Text weight
+ *   + align: Text align
+ * - body-styles:
+ *   + color: Text color
+ *   + align: Text align
+ * - sep:
+ *   + width: Separator's width
+ *   + dash: Separator's style (as a 'line' dash style)
+ */
+ #let showybox(
+  frame: (
+    upper-color: black,
+    lower-color: white,
+    border-color: black,
+    footer-color: luma(220),
+    inset: (x:1em, y:.65em),
+    radius: 5pt,
+    width: 1pt,
+    dash: "solid"
+  ),
+  title-style: (
+    color: white,
+    weight: "bold",
+    align: left
+  ),
+  body-style: (
+    color: black,
+    align: left
+  ),
+  footer-style: (
+    color: luma(85),
+    weight: "regular",
```

Comment:
Down in the function, you use `default: bold` for footer weight, so I don't know if you would wish to use `bold` or `regular` as default 🤔

Reply:
Ah, changed it to "regular" later on. This is the reason I think a central store for defaults would be helpful.